src/openai/dto/apiRequest.dto.ts

```diff
@@ -1,5 +1,32 @@
+// if you think it is better to seperate the interfaces into different entity files, please let me know.
+export interface Message {
+    name: string;
+    content: string;
+    role: Role;
+}
+
+export interface Chat {
+    chatId: number;
+    customerId: number;
+    messages: Message[];
+}
+
+export interface User {
+    userId: number;
+    username: string;
+    roles?: Role[];
+}
+
+enum Role {
+    ASSISTANT = 'assistant',
+    USER = 'user',
+    SYSTEM = 'system',
+    FUNCTION = 'function',
+}
+
 export class apiRequestDTO {
-    _id: string;
-    chatId: string;
-    chatConversation: Record<string, any>;
+    chatId: number;
```

Comment:
you dont actuallly need tha chatId, you only need the json we defined in discord

Reply:
done